src/main/java/com/ithaca/book/BookSearch.java

```diff
@@ -0,0 +1,90 @@
+package com.ithaca.book;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by David on 4/28/16.
+ */
+
+@Service
+public class BookSearch {
+
+    @Autowired
+    BookRepository bookRepository;
+
+    public List<Book> search(String title) {
+        List<Book> all = (List<Book>) bookRepository.findAll();
+
+        return search(title, new ArrayList<>(), all);
+    }
+
+    private List<Book> search(String title, List<Book> currentList, List<Book> all) {
+        if (currentList.size() == 10) {
+            return currentList;
+        }
+
+        Integer bestValue = Integer.MAX_VALUE;
+        Book best = null;
+
+        for(Book book : all) {
+            if (!currentList.contains(book)) {
+
+                Integer score = levenshteinDistance(book.getTitle(), title);
```

Comment:
It's a really basic way of doing a search. It checks how close together 2 strings are. The reason is, if someone searches a title, unless it was an exact match we needed some basic way to say which books we have the in the DB.

That being said this is not at all how a search would really work, we needed something quick because implementing search could have been way to big for the project.